File: SakhaTyla.Web.Ng/src/app/book-author-core/book-author.model.ts
import { OrderDirection } from '../core/models/order-direction.model';
import { PageSettings } from '../core/page.model';
import { BookAuthorFilter } from './book-author-filter.model';

export class BookAuthor {
  id: number;
  creationDate: Date;
  modificationDate: Date;
  creationUserId: number;
  modificationUserId: number;
  lastName: string;
  firstName: string;
  middleName: string;
  nickName: string;
}

export class BookAuthorListState {
  pageSize = PageSettings.pageSize;
  pageIndex = 0;
  filter = new BookAuthorFilter();
  orderBy?: string;
  orderDirection?: OrderDirection;
}

File: SakhaTyla.Core/Requests/Public/Books/Models/BookAuthorModel.cs
namespace SakhaTyla.Core.Requests.Public.Books.Models
{
    public class BookAuthorModel
    {
        public BookAuthorModel(string lastName)
        {
            LastName = lastName;
        }

        public int Id { get; set; }

        public string LastName { get; set; }

        public string? FirstName { get; set; }

        public string? MiddleName { get; set; }

        public string GetFullName()
        {
            return LastName + " "
                + (!string.IsNullOrEmpty(FirstName) ? FirstName[0] + "." : "")
                + (!string.IsNullOrEmpty(MiddleName) ? MiddleName[0] + "." : "");
        }
    }
}
